ompi/attribute/attribute.c

```diff
@@ -343,16 +347,22 @@ do { \
     OPAL_THREAD_LOCK(&attribute_lock); \
 } while (0)
 
-
 /*
  * Cases for attribute values
  */
 typedef enum ompi_attribute_translate_t {
     OMPI_ATTRIBUTE_C,
-    OMPI_ATTRIBUTE_FORTRAN_MPI1,
-    OMPI_ATTRIBUTE_FORTRAN_MPI2
+    OMPI_ATTRIBUTE_FORTRAN_INT,
+    OMPI_ATTRIBUTE_FORTRAN_AINT
 } ompi_attribute_translate_t;
 
+#if SIZEOF_INT == OMPI_SIZEOF_FORTRAN_INTEGER
+#define OMPI_ATTRIBUTE_FORTRAN_MPI1 OMPI_ATTRIBUTE_FORTRAN_INT
+#else
+#define OMPI_ATTRIBUTE_FORTRAN_MPI1 OMPI_ATTRIBUTE_FORTRAN_AINT
+#endif
```

Comment:
I don't understand this block.  It is saying that if `sizeof(int) != sizeof(INTEGER)`, then use the sizeof an Aint.

I understand wanting to rename MPI1/MPI2 to more semantic names, but why not rename `OMPI_ATTRIBUTE_FORTRAN_MPI1` --> `OMPI_ATTRIBUTE_FORTRAN_INTEGER`?  That clearly indicates that we're dealing with Fint's (not ints or Aints).